source/dem/particle_particle_broad_search.cc

```diff
@@ -132,6 +132,176 @@ ParticleParticleBroadSearch<dim>::find_particle_particle_contact_pairs(
     }
 }
 
+template <int dim>
+void
+ParticleParticleBroadSearch<dim>::find_particle_particle_contact_pairs(
+  dealii::Particles::ParticleHandler<dim> &particle_handler,
+  DEMContainerManager<dim> &               container_manager,
+  const DisableParticleContact<dim> &      disable_particle_contact_object)
+{
+  // Pre-fetch and clear containers
+  auto &local_contact_pair_candidates =
+    container_manager.local_contact_pair_candidates;
+  auto &ghost_contact_pair_candidates =
+    container_manager.ghost_contact_pair_candidates;
+  auto &cells_local_neighbor_list = container_manager.cells_local_neighbor_list;
+  auto &cells_ghost_neighbor_list = container_manager.cells_ghost_neighbor_list;
+  local_contact_pair_candidates.clear();
+  ghost_contact_pair_candidates.clear();
+
+  // First we handle the local-local candidate pairs
+
+  // Looping over the potential cells which may contain particles.
+  // This includes the cell itself as well as the neighbouring cells that
+  // were identified.
+  // cell_neighbor_list_iterator is [cell_it, neighbor_0_it, neighbor_1_it, ...]
+  for (auto cell_neighbor_list_iterator = cells_local_neighbor_list.begin();
+       cell_neighbor_list_iterator != cells_local_neighbor_list.end();
+       ++cell_neighbor_list_iterator)
+    {
+      // The main cell & its mobility status
+      auto cell_neighbor_iterator = cell_neighbor_list_iterator->begin();
+      unsigned int main_cell_mobility_status =
```

Comment:
Wherever you have check_cell_mobility, you can just use the key->value of the unordered map instead.